mlrun/model_monitoring/model_monitoring_batch.py

```diff
@@ -638,14 +638,35 @@ def run(self):
                 m_fs = fstore.get_feature_set(
                     f"store://feature-sets/{self.project}/monitoring-{serving_function_name}-{model_name}"
                 )
-                df = m_fs.to_dataframe(
-                    start_time=datetime.datetime.now() - datetime.timedelta(hours=1),
-                    end_time=datetime.datetime.now(),
-                    time_column="timestamp",
-                )
+                try:
+                    df = m_fs.to_dataframe(
+                        start_time=datetime.datetime.now()
+                        - datetime.timedelta(hours=1),
+                        end_time=datetime.datetime.now(),
+                        time_column="timestamp",
+                    )
 
-                # continue if no input provided in the previous hour
-                if len(df) == 0:
+                    if len(df) == 0:
+                        logger.warn(
+                            "Not enough model events since the beginning of the batch interval",
+                            parquet_target=m_fs.status.targets[0].path,
+                            endpoint=endpoint_id,
+                            min_rqeuired_events=mlrun.mlconf.model_endpoint_monitoring.parquet_batching_max_events,
+                            start_time=str(
+                                datetime.datetime.now() - datetime.timedelta(hours=1)
+                            ),
+                            end_time=str(datetime.datetime.now()),
+                        )
+                        continue
+
+                # Continue if not enough events provided since the deployment of the model endpoint
+                except FileNotFoundError:
+                    logger.warn(
+                        "Parquet not found, probably due to not enough model events",
+                        parquet_target=m_fs.status.targets[0].path,
+                        endpoint=endpoint_id,
+                        min_rqeuired_events=mlrun.mlconf.model_endpoint_monitoring.parquet_batching_max_events,
+                    )
```

Comment:
I think you should check the number of model events and in case that actually it's not enough than just log message else fail because this is an error. Where the amount of model events is saved?

Reply:
Talked offline with @yaronha this will require a big change in ParquetTarget (and other targets) to update the status on stream writing and will require a new design.
Decided to leave the workaround for now.